src/statistics.jl

```diff
@@ -500,67 +500,220 @@ function apply_statistic(stat::Density2DStatistic,
     apply_statistic(ContourStatistic(levels=stat.levels), scales, coord, aes)
 end
 
-
+"""
+    A general statistic for density plots (e.g. KDE plots and violin plots).
+See [`Geom.density`](@ref Gadfly.Geom.density) or [`Geom.violin`](@ref
+Gadfly.Geom.violin) for more details.
+"""
```

Comment:
in #1116 i purposely tried to teach new users the GoG way of thinking by making the Geom docstrings explicitly defer explanation of details to the corresponding Stat docstrings for "[derived](https://github.com/GiovineItalia/Gadfly.jl/blame/master/docs/src/lib/geometries.md#L20)" geometries, not the other way around as you've done here.  i feel this is important as it really is a different way of doing things that many find hard to grasp.  does that make sense?

also, it would be good to be explicit about which aesthetics this statistic transforms, again so that the flow of data into making a graph is transparent.  so in this case `x` is transformed into `x` and `y` and is grouped by `color`.  in #1116 i defined [`aes2str`](https://github.com/GiovineItalia/Gadfly.jl/blob/master/src/Gadfly.jl#L72), to which one can input the output of `{input,output}_aesthetics()`,  to help ensure that docs don't get out of sync with code.  might not be useful in this case because of the grouping aesthetic in `:color`.

Reply:
Makes sense. I'll correct this.